src/robusta/core/sinks/robusta/rrm/prometheus_alert_resource_manager.py

```diff
@@ -0,0 +1,209 @@
+            # Only raise an error if it's not a 404 exception.
+            if not (isinstance(e, exceptions.ApiException) and e.status == 404):
+                raise f"An error occurred while fetching PrometheusRules CRD alert: {e}"
+
+        try:
+            # create new if not found
+            if crd_obj is None:
+                self.__write_first_rule(slot, rule)
+            else:
+                crd_obj["spec"]["groups"][0]["rules"].append(rule)
+                self.__k8_api.replace_namespaced_custom_object(
+                    **CRD_PARAMS,
+                    name=name,
+                    body=crd_obj,
+                )
+            self.__cdr_slots_len[slot] += 1
+        except Exception as e:
+            raise f"An error occured while replacing PrometheusRules CRD alert => entity: {resource}. Error: {e}"
+
+    # modify and apply rules to crd file
+    def __modify_rule(self, slot: int, resource: AccountResource):
+        try:
+            name = PrometheusAlertResourceManager.__crd_name(slot)
+
+            crd_obj = self.__k8_api.get_namespaced_custom_object(
+                **CRD_PARAMS,
+                name=name,
+            )
+
+            index = index_of(crd_obj["spec"]["groups"][0]["rules"],
+                             lambda r: r["labels"]["entity_id"] == resource.entity_id)
+            if index >= len(crd_obj["spec"]["groups"][0]["rules"]):
+                return
+
+            rule = resource.resource_state['rule']
+            if "labels" not in rule:
+                rule["labels"] = {}
+            rule["labels"]["entity_id"] = resource.entity_id
+            crd_obj["spec"]["groups"][0]["rules"][index] = rule
+
+            self.__k8_api.replace_namespaced_custom_object(
+                **CRD_PARAMS,
+                name=name,
+                body=crd_obj,
+            )
+        except Exception as e:
+            raise f"An error occured while modify PrometheusRules CRD alert => entity: {resource}. Error: {e}"
+
+    def __write_first_rule(self, slot: int, rule):
+        try:
+            self.__k8_api.create_namespaced_custom_object(
+                **CRD_PARAMS,
+                body=PrometheusAlertResourceManager.__create_prometheus_rule_template(slot, [rule]),
+            )
+        except Exception as e:
+            raise f"An error occured while writing first rule to PrometheusRules CRD alert => rule: {rule}. Error: {e}"
+
+    # delete an existing rule from the crd file
+    def __delete_rule(self, slot, resource: AccountResource):
+        try:
+            name = PrometheusAlertResourceManager.__crd_name(slot)
+
+            crd_obj = self.__k8_api.get_namespaced_custom_object(
+                **CRD_PARAMS,
+                name=name,
+            )
+
+            index = index_of(crd_obj["spec"]["groups"][0]["rules"],
+                             lambda r: r["labels"]["entity_id"] == resource.entity_id)
```

Comment:
I think it may fail if one of the rules doesn't have a `label` for some reason.  I know it is a corner case. It can happen only if someone modifies a CRD. I think later or sooner we will need to handle it as well

Reply:
done.